pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/AuthenticationTest.java

```diff
@@ -1,110 +1,182 @@
 import org.broadinstitute.dsm.model.patch.Patch;
+import org.broadinstitute.dsm.service.admin.SetUserRoleRequest;
+import org.broadinstitute.dsm.service.admin.UserAdminService;
+import org.broadinstitute.dsm.service.admin.UserAdminServiceTest;
 import org.broadinstitute.dsm.statics.DBConstants;
 import org.broadinstitute.dsm.util.UserUtil;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Ignore;
-import org.junit.Test;
+import org.junit.*;
 
-public class AuthenticationTest {
-    public static final String UNIT_TESTER_EMAIL = "testUsersEmails.unitTesterEmail";
-    public static final String GP_UNIT_TESTER_EMAIL = "testUsersEmails.gpUnitTesterEmail";
+import java.util.*;
 
+import static org.broadinstitute.dsm.service.admin.UserAdminService.USER_ADMIN_ROLE;
+import static org.broadinstitute.dsm.statics.DBConstants.KIT_SHIPPING;
+import static org.broadinstitute.dsm.statics.DBConstants.PT_LIST_VIEW;
 
-    @Before
-    public void first() {
-        setupDB();
+public class AuthenticationTest extends DbTxnBaseTest {
 
+    static UserDto dsmAdminUser;
+
+    static UserDto cmiKitShippingOnlyUser;
+
+    static DDPInstanceDto ddpInstance;
+
+    static UserAdminService adminServiceForDsmAdminUser;
+
+    static UserAdminService adminServiceForKitShippingUser;
+    static int pecgsStudyGroupId;
+    static int cmiStudyGroupId;
+
+    static UserDao userDao = new UserDao();
+
+    static final Set<Integer> groupRoleIdsToDelete = new HashSet<>();
+
+    private static UserDto generateUser() {
+        UserDto user = new UserDto("AuthTest." + System.currentTimeMillis(), "fake+" + System.currentTimeMillis() + "@broad.dev", "5555551212");
+        int userId = userDao.create(user);
+        user.setId(userId);
+        return user;
     }
-    @Test
-    @Ignore
-    public void GPPatchKitAccessTest(){
-        String realm = "osteo2";
 
-        String gpUserId = Integer.toString(new UserDao().getUserByEmail(cfg.getString(GP_UNIT_TESTER_EMAIL)).orElse(new UserDto()).getId());
+    /**
+     * Create new groups, a new ddp_instance, associated roles,
+     * a few new users and set their permissions.
+     */
+    @BeforeClass
+    public static void setup() {
+        dsmAdminUser = generateUser();
+        cmiKitShippingOnlyUser = generateUser();
 
-        Patch patch = new Patch("0",  "dsmKitRequestId", "0", gpUserId , new NameValue("kit.collectionDate",  "2023-04-24"), null, "XSZSRS1MS3D4OAEK2DPM") ;
-        patch.setTableAlias("kit");
+        String nameAppend = "." + System.currentTimeMillis();
+        String pecgsStudyGroup = "pecgs" + nameAppend;
+        String studyInstanceName = "instance." + nameAppend;
+        pecgsStudyGroupId = UserAdminService.addStudyGroup(pecgsStudyGroup);
+
+        String cmiStudyGroup = "cmi" + nameAppend;
+        cmiStudyGroupId = UserAdminService.addStudyGroup(cmiStudyGroup);
+
+        int ddpInstanceId = UserAdminServiceTest.createTestInstance(studyInstanceName, cmiStudyGroupId);
```

Comment:
I think with the right test helper class this will be a lot cleaner.

Reply:
Agreed.  Do you have a suggestion for what existing helper class these should be added to, or what a new helper class should be and where it should be?  Some of the methods I've made public in this PR will be needed in non-test code, so I'm thinking about a utility method that is available outside of tests.

Reply:
I built one yesterday. Should be available this morning.

Reply:
It feels like we did not connect well on this? I was assuming you would switch to using src/test/java/org/broadinstitute/dsm/service/admin/UserAdminTestUtil.java which take care of managing test resources for test classes. It has the benefit of not requiring changes to the UserAdminService service contract. (All those UserAdminService methods that were not intended to be public.)